drift-idl-generator/src/main/java/com/facebook/drift/idl/generator/ThriftIdlGenerator.java

```diff
@@ -129,6 +131,46 @@ public ThriftIdlGenerator(ThriftIdlGeneratorConfig config, ClassLoader classLoad
 
         this.namespaces = config.getNamespaces();
         this.recursive = config.isRecursive();
+
+        for (String customCodecClassName : config.getCustomCodecs()) {
+            try {
+                Class<?> codecClass = load(customCodecClassName);
+                if (codecClass == null) {
+                    continue;
+                }
+                Object codec;
+
+                try {
+                    codec = codecClass.getDeclaredConstructor(ThriftCatalog.class).newInstance(catalog);
```

Comment:
Is codec guaranteed to provide such constructor? How does it work for codecs that may need some additional objects to be constructed?